tests/governance_js.py

```diff
@@ -507,22 +508,43 @@ def test_apply(network, args):
             "/gov/proposals",
             proposal(action("always_throw_in_apply")),
         )
-        assert r.status_code == 200, r.body.text()
-        assert r.body.json()["state"] == "Failed", r.body.json()
+        assert r.status_code == 500, r.body.text()
+        assert r.body.json()["error"]["code"] == "InternalError", r.body.json()
         assert (
-            r.body.json()["failure"]["reason"]
+            r.body.json()["error"]["message"].split("\n")[0]
             == "Failed to apply(): Error: Error message"
         ), r.body.json()
 
+    with node.client(None, "member0") as c:
+        pprint.pprint(
+            proposal(action("always_accept_noop"), action("always_throw_in_apply"))
+        )
+        r = c.post(
+            "/gov/proposals",
+            proposal(action("always_accept_noop"), action("always_throw_in_apply")),
+        )
+        assert r.status_code == 200, r.body().text()
+        proposal_id = r.body.json()["proposal_id"]
+        r = c.post(f"/gov/proposals/{proposal_id}/ballots", ballot_yes)
+        assert r.status_code == 200, r.body().text()
+
+        with node.client(None, "member1") as c:
+            r = c.post(f"/gov/proposals/{proposal_id}/ballots", ballot_yes)
+            assert r.body.json()["error"]["code"] == "InternalError", r.body.json()
+            assert (
+                r.body.json()["error"]["message"].split("\n")[0]
+                == "Failed to apply(): Error: Error message"
+            ), r.body.json()
```

Comment:
Can we check that the proposal is reported to be in the `Open` state here?

Reply:
We could, but I think we should not. That transactions returning a non 2xx aren't applied to the KV is a fundamental feature, checking that one value wasn't affected is neither a complete test for that, nor something we want to do every time we return a non-2xx.